### PR TITLE
Support docker in image builder

### PR DESCRIPTION
**Pull request by alexander-demicev**
**What this PR does / why we need it**:
Docker image can now be built with PR and I already published one image ghcr.io/rancher-sandbox/opensuse-rke2:v1.26.4
Ensure that all cloud init services are running and set `ERROR_ON_USER_DATA_FAILURE = False` without it instance fails to bootstrap.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits into logical changes
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

File: image-builder/Makefile [hunk 1]
```diff
@@ -75,3 +80,5 @@ $(AWS_BUILD_TARGETS):
 build-aws-opensuse-1504: ## Builds an AMI for OpenSUSE 15.04
 build-aws-ubuntu-2204: ## Builds an AMI for Ubuntu 22.04
 build-aws-all: $(AWS_BUILD_TARGETS) ## Builds all AMIs
+build-docker-all: $(DOCKER_BUILD_TARGETS) ## Builds all AMIs
+build-docker-opensuse-1504: ## Builds an Docker image for OpenSUSE 15.04
```
Comment (furkatgofurov7) [inline image-builder/Makefile:84]:
```suggestion
build-aws-all: $(AWS_BUILD_TARGETS) ## Builds all AWS AMIs
build-docker-all: $(DOCKER_BUILD_TARGETS) ## Builds all Docker AMIs
build-docker-opensuse-1504: ## Builds a Docker image for OpenSUSE 15.04
```

File: image-builder/aws/packer.json [hunk 2]
```diff
@@ -72,22 +75,21 @@
...
-                "sudo bash ~/bootstrap.sh $CONTAINERD_VERSION $ARCH $RKE2_VERSION",
+                "sudo bash ~/bootstrap.sh $RKE2_VERSION $INFRA",
                 "rm ~/bootstrap.sh"
             ]
         }
     ],
     "variables": {
         "ami_description": "Cluster API base image designed for RKE2 {{user `rke2_semver`}}",
         "ami_groups": "all",
-        "ami_regions": "ap-south-1,eu-west-3,eu-west-2,eu-west-1,ap-northeast-2,ap-northeast-1,sa-east-1,ca-central-1,ap-southeast-1,ap-southeast-2,eu-central-1,us-east-1,us-east-2,us-west-1,us-west-2",
```
Comment (furkatgofurov7) [inline image-builder/aws/packer.json:90]: why are these not needed anymore?

File: image-builder/aws/packer.json [hunk 3]
```diff
@@ -72,22 +75,21 @@
         {
             "environment_vars": [
                 "BUILD_NAME={{user `build_name`}}",
-                "CONTAINERD_VERSION={{user `containerd_version`}}",
-                "ARCH=amd64",
```
Comment (furkatgofurov7) [inline image-builder/aws/packer.json:76]: is this a containerd specific var?

**Comment by richardcase** [comment]
Sorry @alexander-demicev this dropped off my radar. I will take a look

**Comment by alexander-demicev** [comment]
Closing it for now, I had some issues with enabling systemd here. Will reopen if needed.

**Pull request closed without merging**